### ISSUE TITLE
Network Error logic obscures response (HTTP 401 mishandled in IE 10)

### ISSUE THREAD

**Issue by noahmonster** [mzabriskie/axios#566]
Hi,

Let me just say upfront that this is a request to work around something that is broken in IE 10.

Background information - [IE 10 has a bug where it treats HTTP 401 and 403 as Network Errors.](https://connect.microsoft.com/IE/feedback/details/785990/ie-10-on-win8-does-not-assign-the-correct-status-to-xmlhttprequest-when-the-result-is-401)  The resulting behavior is that the `status` is set to 0, which causes Axios to treat the response as a "low level network error"

Unfortunately, Axios' [low level network error handler](https://github.com/mzabriskie/axios/blob/417913bf8b0b8c3530e3df122b4581cd83f0feb5/lib/adapters/xhr.js#L84) neglects to set the response object from the server on the rejected Promise.  For "normal" Network Errors, there would not be a response object from the server, but in this particular case there is.  Having a `response` set on this error and being able to inspect it would be the only way to differentiate these peculiar cases in IE.

**TLDR**; we should be setting a `response` on Network Errors if it is available.

**Comment by noahmonster** [comment]
After doing more research it has become apparent that IE is actually hiding the response even though the browser is receiving it.  Looks like there is nothing that can be done.

**Comment by conanmy** [comment]
I am encountering same issue on Chrome Version 64.0.3282.140 (Official Build) (64-bit).
